pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -263,6 +243,10 @@ def get_mp_spawn_kwargs(self, trainer: Optional["pl.Trainer"] = None) -> Dict[st
         }
 
     def spawn(self, function: Callable, *args: Any, **kwargs: Any) -> Optional[Union[Any, "_SpawnOutput"]]:
+        # TODO: this todo is unclear, does it still apply?
+        # todo: precision pluging is call in accelerator setup and should be moved
+        if "XLA_USE_BF16" in os.environ:
+            del os.environ["XLA_USE_BF16"]
```

Comment:
n00b question: why is this key removed? 

before this was removed only before training, but not evaluation/prediction. was that a bug in the previous code that this is fixing?

Reply:
True, something is odd here. Actually, I don't think the variable should be removed at all. It is being set initially in the precision plugin connect() method, so removing it here right after makes no sense. I will address this in #10990.

Reply:
Ok, as discussed in #10990, I will carry this bug over in this PR and fix it by teardown() in the linked PR. Resolving